exercises/handling-errors/solution/src/workflows.ts

```diff
@@ -0,0 +1,62 @@
+import { proxyActivities, ApplicationFailure, ActivityFailure, log, sleep } from '@temporalio/workflow';
+import type * as activities from './activities';
+import { Distance, OrderConfirmation, PizzaOrder} from './shared';
+
+const { sendBill, getDistance, validateCreditCard } = proxyActivities<typeof activities>({
+  startToCloseTimeout: '5 seconds',
+});
+
+export async function pizzaWorkflow(order: PizzaOrder): Promise<OrderConfirmation> {
+  let totalPrice = 0;
+
+  // Validate the credit card number
+  try {
+    await validateCreditCard(order.customer.creditCardNumber);
+  } catch (err) {
+    if (err instanceof ActivityFailure && err.cause instanceof ApplicationFailure) {
+      log.error(err.cause.message);
+    } else {
+      log.error(`error validating credit card number: ${err}`);
+    }
+  }
+
+  if (order.isDelivery) {
+    let distance: Distance | undefined = undefined;
+
+    try {
+      distance = await getDistance(order.address);
+    } catch (err) {
+      log.error('Unable to get distance', {});
+      throw err;
+    }
+
+    if (distance.kilometers > 25) {
+      log.error(`Customer lives too far away: ${distance.kilometers} km`);
+      throw ApplicationFailure.create({
+        message: 'Customer lives too far away for delivery',
+        details: [distance.kilometers],
+      });
+    }
+  }
+
+  for (const pizza of order.items) {
+    totalPrice += pizza.price;
+  }
+
+  // We use a short Timer duration here to avoid delaying the exercise
+  await sleep('3 seconds');
+
+  const bill = {
+    customerID: order.customer.customerID,
+    orderNumber: order.orderNumber,
+    amount: totalPrice,
+    description: 'Pizza',
+  };
+
+  try {
+    return await sendBill(bill);
+  } catch (e) {
+    log.error('Unable to bill customer', {});
```

Comment:
Why the empty object? Don't you want to include the error message? i.e. `log.error(`Unable to bill customer: ${e}`)`